### PR TITLE
Balder/gcc 8 prep 1

### PR DESCRIPTION
**Pull request by baldersheim**
@toregge PR

### CODE REVIEW

**Comment by baldersheim** [comment]
@toregge This is an initial take at some of the errors,more fixes will come in later pull requests

File: vespamalloc/src/vespamalloc/malloc/memorywatcher.h [hunk 1]
```diff
@@ -231,8 +231,8 @@ const char *vespaHomeConf(char pathName[])
     if (env != NULL) {
         home = env;
     }
-    strncpy(pathName, home, PATH_MAX);
-    strncat(pathName, "/etc/vespamalloc.conf", PATH_MAX);
+    strncpy(pathName, home, PATH_MAX-1);
+    strncat(pathName, "/etc/vespamalloc.conf", PATH_MAX-1 -strlen(pathName));
```
Comment (toregge) [inline vespamalloc/src/vespamalloc/malloc/memorywatcher.h]: What happens if strlen(home) >= PATH_MAX ?

File: vespalog/src/vespa/log/log.cpp [hunk 2]
```diff
@@ -128,7 +128,7 @@ Logger::Logger(const char *name, const char *rcsId)
     memset(_appendix, 0, sizeof(_appendix));
     const char *app(strchr(name, '.') ? strchr(name, '.') : "");
     assert(strlen(app) < sizeof(_appendix));
-    strncpy(_appendix, app, sizeof(_appendix));
+    strncpy(_appendix, app, sizeof(_appendix)-1);
```
Comment (toregge) [inline vespalog/src/vespa/log/log.cpp]: strncpy might not NUL-terminate _appendix

File: fastos/src/tests/tests.h [hunk 3]
```diff
@@ -43,7 +43,7 @@ class BaseTest : public FastOS_Application
    bool Progress (bool result, const char *str)
    {
       char string[MAX_STR_LEN];
-      sprintf(string, "%s: %s\n",
+      snprintf(string, sizeof(string) - 1, "%s: %s\n",
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?. The size passed to snprintf includes the NUL byte.

File: fastos/src/tests/tests.h [hunk 4]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
 
    bool Progress (bool result, const char *str, int d1)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, d1);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, d1);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 5]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, int d1, int d2)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, d1, d2);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, d1, d2);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 6]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, const char *s1)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, s1);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, s1);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 7]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, const FastOS_ThreadInterface *s1)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, s1);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, s1);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 8]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, const FastOS_Socket *s1)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, s1);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, s1);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 9]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, const char *s1, const char *s2)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, s1, s2);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, s1, s2);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 10]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, const char *s1, int d1)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, s1, d1);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, s1, d1);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 11]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool Progress (bool result, const char *str, int d1, const char *s1)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, d1, s1);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, d1, s1);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 12]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool ProgressI64 (bool result, const char *str, int64_t val)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, val);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, val);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: fastos/src/tests/tests.h [hunk 13]
```diff
@@ -52,71 +52,71 @@ class BaseTest : public FastOS_Application
...
       return Progress(result, string);
    }
 
    bool ProgressFloat (bool result, const char *str, float val)
    {
-      char string[MAX_STR_LEN];
-      sprintf(string, str, val);
+      char string[MAX_STR_LEN-100];
+      snprintf(string, sizeof(string) - 1, str, val);
```
Comment (toregge) [inline fastos/src/tests/tests.h]: Why -1 ?

File: vespamalloc/src/vespamalloc/malloc/memorywatcher.h (hunk 1 above)
Comment (baldersheim) [reply vespamalloc/src/vespamalloc/malloc/memorywatcher.h]: Good point, asserting instead.

File: vespalog/src/vespa/log/log.cpp (hunk 2 above)
Comment (baldersheim) [reply vespalog/src/vespa/log/log.cpp]: Fixed

File: fastos/src/tests/tests.h (hunk 3 above)
Comment (baldersheim) [reply fastos/src/tests/tests.h]: That was to avoid warning. But as you point out the ensuring of zero termination was missing.
Added.

**Comment by baldersheim** [comment]
@toregge Now I think it should be ready.

File: vespamalloc/src/vespamalloc/malloc/memorywatcher.h [hunk 14]
```diff
@@ -227,12 +227,14 @@ namespace {
...
 {
     const char *home = "/opt/vespa";
+    const char *conf = "/etc/vespamalloc.conf";
     const char *env = getenv("VESPA_HOME");
     if (env != NULL) {
         home = env;
     }
-    strncpy(pathName, home, PATH_MAX-1);
-    strncat(pathName, "/etc/vespamalloc.conf", PATH_MAX-1 -strlen(pathName));
+    assert((strlen(home)  + strlen(conf) + 1) < PATH_MAX);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply vespamalloc/src/vespamalloc/malloc/memorywatcher.h]: Used explicit if/ abort

**Comment by baldersheim** [comment]
@vekterli FYI

**Review by toregge — COMMENTED** [review]
LGTM

**Pull request merged**